test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_update_state.cpp

```diff
@@ -0,0 +1,345 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "testHarness.h"
+#include "mutable_command_basic.h"
+
+#include <CL/cl_ext.h>
+
+#include <vector>
+
+namespace {
+
+////////////////////////////////////////////////////////////////////////////////
+// Tests related to ensuring the state of the updated command-buffer is expected
+// and the effects of operations on it don't have side effects on other objects.
+//
+// - Tests the updates applied to a command-buffer persist over all subsequent
+// enqueues.
+// - Tests interaction of `clSetKernelArg` with mutable-dispatch extension.
+
+struct MutableDispatchUpdateStateTest : public BasicMutableCommandBufferTest
+{
+    MutableDispatchUpdateStateTest(cl_device_id device, cl_context context,
+                                   cl_command_queue queue)
+        : BasicMutableCommandBufferTest(device, context, queue),
+          buffer(nullptr), command(nullptr)
+    {}
+
+    bool Skip() override
+    {
+        if (BasicMutableCommandBufferTest::Skip()) return true;
+
+        cl_mutable_dispatch_fields_khr mutable_capabilities;
+        bool mutable_support =
+            !clGetDeviceInfo(
+                device, CL_DEVICE_MUTABLE_DISPATCH_CAPABILITIES_KHR,
+                sizeof(mutable_capabilities), &mutable_capabilities, nullptr)
+            && mutable_capabilities & CL_MUTABLE_DISPATCH_ARGUMENTS_KHR;
+        return !mutable_support;
+    }
+
+    cl_int SetUpKernelArgs() override
+    {
+        cl_int error = CL_SUCCESS;
+        buffer = clCreateBuffer(context, CL_MEM_READ_WRITE,
+                                num_elements * sizeof(cl_int), nullptr, &error);
+        test_error(error, "clCreateBuffer error");
+
+        // Zero initialize buffer
+        const cl_int zero_pattern = 0;
+        error = clEnqueueFillBuffer(
+            queue, buffer, &zero_pattern, sizeof(cl_int), 0,
+            num_elements * sizeof(cl_int), 0, nullptr, nullptr);
+        test_error(error, "clEnqueueFillBuffer failed");
+
+        error = clFinish(queue);
+        test_error(error, "clFinish failed");
+
+        error = clSetKernelArg(kernel, 0, sizeof(buffer), &buffer);
+        test_error(error, "Unable to set kernel argument 0");
+
+        return CL_SUCCESS;
+    }
+
+    cl_int SetUpKernel() override
+    {
+        const char *add_kernel =
+            R"(
+            __kernel void add_kernel(__global int *data, int value)
+            {
+                size_t tid = get_global_id(0);
+                data[tid] += value;
+            })";
+
+        cl_int error = create_single_kernel_helper(
+            context, &program, &kernel, 1, &add_kernel, "add_kernel");
+        test_error(error, "Creating kernel failed");
+        return CL_SUCCESS;
+    }
+
+    cl_int Run() override
```

Comment:
I don't think this version of "Run" from the base class ever gets called - does it?

If not, it can be removed.